### PR TITLE
PLATUI-64: fixed broken tests

### PR DESCRIPTION
**Pull request by silasl**
* Re-implemented skipped tests
* Fixed implementation of Welsh language option and added new test

### CODE REVIEW

File: application/__tests__/example.macro.test.js [hunk 1]
```diff
@@ -68,100 +82,39 @@ xdescribe('Single page example macro english html only', () => {
...
-  welshOptions.filters = Object.assign({}, options.filters)
-  welshOptions.globals = Object.assign({ hasWelsh: true }, welshOptions.globals)
-
-  const parameters = { html: `${exampleId}.html` }
-
-  const document = documentFactory(parameters, welshOptions)
-
-  test('Should include a language toggle for examples', () => {
-    const languageToggleLink = document.querySelector('a.language-toggle')
+  test('Should not have a language toggle for dual language examples', () => {
```
Comment (matcarey) [inline application/__tests__/example.macro.test.js]: Is this a stray `not`?

File: application/__tests__/example.macro.test.js [hunk 2]
```diff
@@ -68,100 +82,39 @@ xdescribe('Single page example macro english html only', () => {
...
-  test('Should have a Welsh Language example version in a hidden div', () => {
-    const exampleHTMLCode = document.getElementById(`${exampleId}_welsh`)
-    expect(exampleHTMLCode.classList).toContain('govuk-visually-hidden')
-    expect(exampleHTMLCode.getAttribute('aria-hidden')).toBe('true')
-
-    const fixtureHTML = fs.readFileSync(path.join(fixturePath, 'test.cy.html')).toString()
-    expect(exampleHTMLCode.querySelector('pre code').innerHTML).toEqual(htmlEscape(fixtureHTML))
+  test('should have a button to show HTML code examples', () => {
+    const tabLink = document.querySelector('ul.app-tabs li.js-tabs__item a[href="#example-default-html"]')
+    const tabContentContainer = document.getElementById('example-default-html')
```
Comment (matcarey) [inline application/__tests__/example.macro.test.js]: Is it worth variablising `'example-default-html'` to show/ensure that it's the same value used multiple times?  (`aria-controls` is the other time).

File: application/__tests__/example.macro.test.js [hunk 3]
```diff
@@ -68,100 +82,39 @@ xdescribe('Single page example macro english html only', () => {
...
-
-  test('should set the iframe to use the first example page when the html parameter is an Array', () => {
-    expect(exampleFrame.name).toBe('example1_frame')
-    expect(exampleFrame.src).toBe('examples' + '/' + parameters.html[0])
-  })
-  test('should not include the example code buttons by default', () => {
-    expect(document.getElementsByClassName('app-tabs__item').length).toBe(0)
+  test('should have a button to show Nunjucks code examples', () => {
+    const tabLink = document.querySelector('ul.app-tabs li.js-tabs__item a[href="#example-default-nunjucks"]');
+    const tabContentContainer = document.getElementById('example-default-nunjucks')
```
Comment (matcarey) [inline application/__tests__/example.macro.test.js]: (If we're variablising for `example-default-html` we should do the same here)

File: lib/filters.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+const path = require('path')
+const highlight = require('./highlighter')
+
+module.exports = {
+  is_array: function (obj) { return Array.isArray(obj) },
+  dirname: function (filepath) { return path.dirname(filepath) },
```
Comment (matcarey) [inline lib/filters.js]: As we're starting to use arrow functions elsewhere should we update these too?

File: lib/globals.js [hunk 5]
```diff
@@ -26,6 +20,7 @@ exports.getHTMLCode = path => {
     html = nunjucks.renderString(content)
   } catch (err) {
     if (err) {
+      console.log(err)
```
Comment (matcarey) [inline lib/globals.js]: `console.error` should be safe to use here.

File: package.json [hunk 6]
```diff
@@ -5,8 +5,9 @@
   "description": "Static site generator for the HMRC Design System",
   "scripts": {
     "start": "gulp",
-    "test": "gulp lint && jest",
-    "build": "npm run test && gulp build",
+    "test": "gulp lint && jest --watch",
```
Comment (matcarey) [inline package.json]: Can we switch round the watch task to a different name?  That way:

`npm test` will give a status code result
`npm run test:watch` or `npm run dev` will support developers

I like the idea that `npm test` is a consistent feature across projects, I think that consistency should be available on build servers etc.

File: lib/globals.js (hunk 5 above)
Comment (silasl) [reply lib/globals.js]: I've removed this completely as it was left over from debugging and don't necessarily wont to be logging errors in prod without impact assessment?

Comment (silasl) [reply lib/globals.js]: Edit: reintroduced as console.error as discussed

**Pull request merged**